src/components/util/dnd-table.js

```diff
@@ -0,0 +1,270 @@
+    }
+
+    //TODO fix alignment of column names when no rows ?
+    function renderTableHead() {
+        return (
+            <TableHead>
+                <TableRow>
+                    <TableCell>
+                        {/* empty cell for the drag and drop column */}
+                    </TableCell>
+                    <TableCell>
+                        <MultiCheckbox
+                            arrayFormName={arrayFormName}
+                            handleClickIfChecked={selectAllRows}
+                            handleClickIfUnchecked={unselectAllRows}
+                        />
+                    </TableCell>
+                    {columnsDefinition.map((column) => (
+                        <TableCell key={column.dataKey}>
+                            {column.label}
+                            {column.extra}
+                        </TableCell>
+                    ))}
+                </TableRow>
+            </TableHead>
+        );
+    }
+
+    function renderTableBody(providedDroppable) {
+        return (
+            <TableBody>
+                {tapFields.map((row, index) => (
+                    <Draggable
+                        key={row.id}
+                        draggableId={row.id.toString()}
+                        index={index}
+                    >
+                        {(provided, snapshot) => (
+                            <TableRow
+                                ref={provided.innerRef}
+                                {...provided.draggableProps}
+                            >
+                                <Tooltip
+                                    title={intl.formatMessage({
+                                        id: 'DragAndDrop',
+                                    })}
+                                    placement="right"
+                                >
+                                    <TableCell {...provided.dragHandleProps}>
+                                        <DragIndicatorIcon />
+                                    </TableCell>
+                                </Tooltip>
+                                <TableCell>
+                                    <CheckboxInput
+                                        name={`${arrayFormName}[${index}].${SELECTED}`}
+                                    />
+                                </TableCell>
+                                {columnsDefinition.map((column) =>
+                                    renderTableCell(row.id, index, column)
+                                )}
+                            </TableRow>
+                        )}
+                    </Draggable>
+                ))}
+                {providedDroppable.placeholder}
+            </TableBody>
+        );
+    }
+
+    return (
+        <Grid item container spacing={1}>
+            <Grid item container>
+                <DragDropContext onDragEnd={onDragEnd}>
+                    <Droppable droppableId="tapTable">
```

Comment:
is the Droppable area not too large ?

Reply:
I don't think so, then you can use the header area too. It's like that in th documentation too.
https://github.com/atlassian/react-beautiful-dnd/blob/master/docs/patterns/tables.md
https://react-beautiful-dnd.netlify.com/?selectedKind=Tables&selectedStory=with%20fixed%20width%20columns&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel